src/deluge/model/sample/sample_recorder.cpp

```diff
@@ -1049,9 +1095,23 @@ void SampleRecorder::feedAudio(int32_t* __restrict__ inputAddress, int32_t numSa
 				} while (inputAddress < endInputNow);
 			}
 
-			writePos = writePosNow;
-
-			numSamplesCaptured += numSamplesThisCycle;
+			// if we're not threshold recording or we're threshold recording and detected audio
+			if (!thresholdRecording || sample->audioStartDetected) {
+				writePos = writePosNow;
+				numSamplesCaptured += numSamplesThisCycle;
+			}
+			// if we're threshold recording and didn't detect audio in previous cycles
+			// check if there's any audio in this cycle
+			else {
+				StereoFloatSample approxRMSLevel =
+				    envelopeFollower.calcApproxRMS((StereoSample*)beginInputNow, numSamplesThisCycle);
+				if (std::max(approxRMSLevel.l, approxRMSLevel.r) > startValueThreshold) {
+					writePos = writePosNow;
+					numSamplesCaptured += numSamplesThisCycle;
+					sample->audioStartDetected = true;
+					envelopeFollower = {};
```

Comment:
I think there's a reset method, it's just to set its current level to zero. Otherwise it would still be high when the next sample is started

Reply:
> I think there's a reset method, it's just to set its current level to zero. Otherwise it would still be high when the next sample is started

mm I don't see a reset method in the AbsValueFollower class

I see a setup function, would you just feed 0 to that?

// takes in knob positions in the range 0-ONE_Q31
void AbsValueFollower::setup(q31_t a, q31_t r) {
	setAttack(a);
	setRelease(r);
}

Reply:
Oops, guess it didn't make it over from the compressor code. it's just something like this 

``` 
	/// Reset the state of the follower
	void reset() {
		lastMeanl = 0;
		lastMeanr = 0;
	}
```

Reply:
> Oops, guess it didn't make it over from the compressor code. it's just something like this
> 
> ```
> 	/// Reset the state of the follower
> 	void reset() {
> 		lastMeanl = 0;
> 		lastMeanr = 0;
> 	}
> ```

perfect, implemented that thanks!